ai/ai_model_building/unit_parser.py

```diff
@@ -1,22 +1,94 @@
 import json
 import os
-from stat_parser import create_stat
+from stat_parser import create_stat, stats_config, custom_load_unit_stat
 import pathlib
+import copy
 
 UNITS_FOLDER = str(pathlib.Path(__file__).parent.resolve()
                    ) + '/units_descriptor'
 
 UNITS_FILES = [file for file in os.listdir(
     UNITS_FOLDER) if file.endswith('.json')]
 
+def build_subunits(file_unit, subunits):
+    """Generate the subunits of a unit
```

Comment:
What is a subunit ?

Reply:
![image](https://github.com/Prophecy-eip/Maths/assets/64773080/4b815b81-bd0e-4105-bfb7-47459688aa0f)
It's those variants you see here: the knight aspirant and the barded warhorse
It was the only way I found so far to create those unit from our jsons